File: ios/App/App/RemotePush.m
//
//  RemotePush.m
//  App
//
//  Created by Joseph Yang on 10/20/21.
//

#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(RemotePush, "RemotePush",
           CAP_PLUGIN_METHOD(NativeMethod, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(NotifyListeners, CAPPluginReturnPromise);
)

File: ios/App/App/RemotePush.swift
//
//  RemotePush.swift
//  App
//
//  Created by Joseph Yang on 10/20/21.
//

import Foundation
import Capacitor

@objc(RemotePush)
public class RemotePush: CAPPlugin {
    
    public override func load() {
        NotificationCenter.default.addObserver(self, selector: #selector(self.handleDataMessage(notification:)), name: Notification.Name("OnRemoteMessage"), object: nil)
    }

    @objc func handleDataMessage( notification: NSNotification) {
        let userInfo: [AnyHashable : Any]
        userInfo = notification.object as! [AnyHashable : Any]
        print("Entire message \(userInfo)")
        self.notifyListeners("OnRemoteNotification", data: ["data":userInfo])
    }
}
